docs/plugins.md

```diff
@@ -0,0 +1,72 @@
+# Plugins
+
+![Plugins](assets/plugins.png)
+
+Plugins allow you to extend Argo Workflows to add new capabilities.
+
+* You don't need to learn Golang, you can write in any language, including Python.
+* Simple: a plugin just responds to RPC HTTP requests.
+* You can iterate quickly by changing the plugin at runtime.
+* You can get your plugin running today, no need to wait 3-5 months for an Argo software release.
+
+[Executor plugins](executor_plugins.md) written and installed by both users and admins.
+
+## Configuration
+
+Plugins are disabled by default. To enable them, start the controller with `ARGO_PLUGINS=true`, e.g.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: workflow-controller
+spec:
+  template:
+    spec:
+      containers:
+        - name: workflow-controller
+          env:
+            - name: ARGO_PLUGINS
+              value: "true"
+```
+
+## Considerations
+
+### Security
+
+No security context is enforced, but is strong encouraged you configure your container not to run as root.
+
+Plugins can run at an non-localhost address, but this is very strongly discouraged. Within a pod, network traffic cannot
+be observed.
```

Comment:
Ditto above. If we're going to support this mode, I think it should be first-class.